framework/include/meshlet_helpers.hpp

```diff
@@ -0,0 +1,174 @@
+
+	/** Divides the given models into meshlets using the given callback function.
+	 *  @param	aModelsAndMeshletIndices				All the models and associated meshes that should be divided into meshlets.
+	 *  @param	aMeshletDivision	Callback used to divide meshes into meshlets with a maximum number of vertices and indices.
+	 *								It can either receive the vertices and indices or just the indices depending on your specific needs.
+	 *								Additionally it provides the model and an optional mesh index if more data is needed. If no mesh index is provided then the meshes were combined beforehand.
+	 *								Must not grab ownership of the model, the model will be assigned to the meshlets after its execution.\n\n
+	 *								The callback has a specific layout, optional parameters can be omitted, but all of them need to be provided in the following order.
+	 *								Parameters in definition order:
+	 *								 - const std::vector<glm::vec3>& tVertices:		optional	The vertices of the mesh or combined meshes of the model
+	 *								 - const std::vector<uint32_t>& tIndices:  		mandatory	The indices of the mesh or combined meshes of the model
+	 *								 - const model_t& tModel:						mandatory	The model these meshlets are generated from
+	 *								 - std::optional<mesh_index_t> tMeshIndex: 		mandatory	The optional mesh index. If no value is provided, it means the meshes of the model are combined into a single vertex and index buffer
+	 *								 - uint32_t tMaxVertices:						mandatory	The maximum number of vertices that are allowed in a single meshlet.
+	 *								 - uint32_t tMaxIndices:						mandatory	The maximum number of indices that are allowed in a single meshlet.
+	 *								Return value:
+	 *								 - std::vector<meshlet>										The generated meshlets.\n\n
+	 *								Example:
+	 *								@code
+	 *								[](const std::vector<glm::vec3>& tVertices, const std::vector<uint32_t>& aIndices,
+	 *											const model_t& aModel, std::optional<mesh_index_t> aMeshIndex,
+	 *											uint32_t aMaxVertices, uint32_t aMaxIndices) {
+	 *									std::vector<gvk::meshlet> generatedMeshlets;
+	 *									// Do your meshlet division here
+	 *									return generatedMeshlets;
+	 *								}
+	 *								@endcode
+	 *	@param	aMaxVertices		The maximum number of vertices of a meshlet.
+	 *	@param	aMaxIndices			The maximum number of indices of a meshlet.
+	 *	@param	aCombineSubmeshes	If submeshes should be combined into a single vertex/index buffer.
+	 */
+	template <typename F>
+	extern std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModelsAndMeshletIndices, F aMeshletDivision,
+		const uint32_t aMaxVertices = 64, const uint32_t aMaxIndices = 378, const bool aCombineSubmeshes = true)
+	{
+		std::vector<meshlet> meshlets;
+		for (auto& pair : aModelsAndMeshletIndices) {
+			const auto& model = std::get<avk::resource_ownership<model_t>>(pair);
```

Comment:
```suggestion
			auto& model = std::get<avk::resource_ownership<model_t>>(pair);
```

This must not be a const reference, because a const reference cannot be moved from!
This means that the `std::move(model)` further down cannot perform the cast into an r-value reference and instead o being moved, the variable is being copied (probably compiler specific, but `const auto&` is definitely not the way to go here).

Reply:
Ok yeah makes sense. Got a bit confused there from previous comments. 
I'll need to change the `aModelsAndMeshletIndices` to non const now, else it would still be deduced as const according to https://stackoverflow.com/a/32130883.

Reply:
Ah, yeah, sure!